src/ext/java/org/opentripplanner/ext/transmodelapi/mapping/ViaLocationMapper.java

```diff
@@ -1,17 +1,33 @@
 package org.opentripplanner.ext.transmodelapi.mapping;
 
+import static org.opentripplanner.routing.api.response.InputField.INTERMEDIATE_PLACE;
+
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 import org.opentripplanner.routing.api.request.ViaLocation;
+import org.opentripplanner.routing.api.response.RoutingError;
+import org.opentripplanner.routing.api.response.RoutingErrorCode;
+import org.opentripplanner.routing.error.RoutingValidationException;
 
 class ViaLocationMapper {
 
   static ViaLocation mapViaLocation(Map<String, Object> viaLocation) {
+    validateViaLocation(viaLocation);
```

Comment:
This validation rule should be inside the `ViaLocation` constructor, not in the mapper?

To get the error, the constructor should throw a IllegalArgumentException and the `validateViaLocation` should catch the exception and convert it to a `RoutingError`. This would then encapsulate the business rule that either `place` or `coordinates` must be set. The rule should not be implemented in a mapper.

Reply:
Done